src/PROPOSAL/PROPOSAL/propagation_utility/ContRandBuilder.h

```diff
@@ -27,11 +27,11 @@ template <class T> class ContRandBuilder : public ContRand {
     }
 
     inline double EnergyRandomize(
-        double initial_energy, double final_energy, double rnd) final
+        double initial_energy, double final_energy, double rnd, double min_energy = 0) final
     {
         auto std = std::sqrt(Variance(initial_energy, final_energy));
         return SampleFromGaussian(final_energy, std, rnd,
-                                  std::min(disp->GetLowerLim(), final_energy),
+                                  std::max(disp->GetLowerLim(), min_energy),
```

Comment:
So this is the key change of this PR.
Before, the minimum of the lower limit and the final energy was taken. Now, the final energy is not considered anymore. I don't fully understand why this was done in the past. But if there was a good reason for this, maybe an edge case, we must check if this edge case is still treated correctly.

Reply:
and if the sampled final energy is below the lower limit of interpolated displacement energies (which is usually the mass of the particle or a very small energy for massless particles like photons or neutrinos), this raises other issues, I think.
Just wanted to be sure.
For me, the current change makes more sense.
However, maybe a comment could help to solve an issue in the future, in case we overlook something, here.

I would enhance the `Constraints` of the `ContinuousRandomization` UnitTest with a minimal energy. Those events, ending with a stochastic interaction instead of a continuous loss, should of course be avoided as they can be below the threshold, as explained above.